client/engine/store/mocktore_test.go

```diff
@@ -0,0 +1,65 @@
+package store
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/statechannels/go-nitro/channel/state"
+	directfund "github.com/statechannels/go-nitro/protocols/direct-fund"
+	"github.com/statechannels/go-nitro/types"
+)
+
+func TestNewMockStore(t *testing.T) {
+	NewMockStore([]byte{'a', 'b', 'c'})
+}
+
+func TestSetGetObjective(t *testing.T) {
+	ms := NewMockStore([]byte{})
+	got, ok := ms.GetObjectiveById("404")
+	if ok {
+		t.Errorf("expected not to find the 404 objective, but found %v", got)
+	}
+
+	ts := state.TestState
+	ts.TurnNum = 0
+
+	testObj, _ := directfund.New(
+		ts,
+		ts.Participants[0],
+		false,
+		types.AddressToDestination(ts.Participants[0]),
+		types.AddressToDestination(ts.Participants[1]),
+	)
+
+	ms.SetObjective(testObj)
```

Comment:
<img width="868" alt="Screen Shot 2022-01-14 at 4 00 35 PM" src="https://user-images.githubusercontent.com/8432675/149599821-69b9e4c9-6e8c-4e80-aded-c59df820f711.png">

This is such a nice feature.